newIDE/app/src/MainFrame/index.js

```diff
@@ -1163,16 +1166,19 @@ class MainFrame extends React.Component<Props, State> {
   };
 
   save = () => {
+    this.setState({
+      isSaveAs: false,
+    });
     saveUiSettings(this.state.editorTabs);
 
-    const { currentProject } = this.state;
+    const { currentProject, isSaveAs } = this.state;
     if (!currentProject) return;
     const { i18n } = this.props;
 
     if (this.props.saveDialog) {
       this._openSaveDialog();
     } else if (this.props.onSaveProject) {
-      this.props.onSaveProject(currentProject).then(
+      this.props.onSaveProject(currentProject, isSaveAs).then(
```

Comment:
~~Just pass false instead of a variable~~

In fact, don't change this function. We'll make a new function called `onSaveProjectAs`. See the other comment